model/src/main/java/com/flipkart/flux/domain/Event.java

```diff
@@ -13,63 +13,123 @@
 
 package com.flipkart.flux.domain;
 
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.Type;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import javax.persistence.*;
 import java.io.Serializable;
+import java.util.Date;
 
 /**
  * <code>Event</code> is the result of a {@link Task} execution.
  * It is to be posted back to the Flux execution engine once a worker has executed the task
- * 
+ *
  * @author Yogesh
  * @author regunath.balasubramanian
  * @author shyam.akirala
  */
+
+@Entity
 public class Event<T> implements Serializable {
-	
-	/** Default serial version UID*/
-	private static final long serialVersionUID = 1L;
-	
-	/** The name for this Event*/
+
+    /** Default serial version UID*/
+    private static final long serialVersionUID = 1L;
+
+    /** Auto generated Id*/
+    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    /** The name for this Event*/
     private String name;
-    
+
     /** The type of this Event*/
     private String type;
-    
-    /** Staus for this Event*/
+
+    /** Status for this Event*/
+    @Enumerated(EnumType.STRING)
     private EventStatus status;
-    
-    /** Data associated with this Event*/
+
+    /** Instance Id of state machine with which this event is associated */
+    private String stateMachineInstanceId;
+
+    /** Data associated with this Event, must have public getters and setters and be serializable */
+    @Type(type = "BlobType")
     private T eventData;
 
+    /** The source who generated this Event */
+    private String eventSource;
+
+    /** Event creation time */
+    @CreationTimestamp
```

Comment:
don't use hibernate to get the creationTimestamp. Let it come from the DB itself.
That way we won't have to bother about time differences between service nodes (if any)